File: app/src/main/java/com/github/fwh007/kotlintest/dagger/MainActivity.java
package com.github.fwh007.kotlintest.dagger;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.widget.Toast;

import javax.inject.Inject;

/**
 * TODO 类说明
 *
 * @author : Winter
 */


public class MainActivity extends AppCompatActivity {
    @Inject
    Pot pot;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        DaggerMainActivityComponent.create().inject(this);
        Toast.makeText(this, pot.show(), Toast.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/com/github/fwh007/kotlintest/MainActivity.kt
package com.github.fwh007.kotlintest

import com.github.fwh007.kotlintest.view.BaseFragment

class MainActivity : android.support.v7.app.AppCompatActivity() {

    override fun onCreate(savedInstanceState: android.os.Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val toolbar = findViewById(R.id.toolbar) as android.support.v7.widget.Toolbar
        setSupportActionBar(toolbar)
    }

    override fun onCreateOptionsMenu(menu: android.view.Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: android.view.MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        val id = item.itemId

        if (id == R.id.action_settings) {
            return true
        }

        return super.onOptionsItemSelected(item)
    }

    fun startFragment(fragment: BaseFragment) {
        supportFragmentManager.beginTransaction()
                .addToBackStack(null)
                .replace(R.id.fl_fragment, fragment, fragment.javaClass.simpleName)
                .commit()
    }

    fun a() = fun(c: Int) = c + 1

    fun <A, B, C> compose(f: (B) -> C, g: (A) -> B): (A) -> C {
        return { x -> f(g(x)) }
    }
}
